### ISSUE TITLE
No experiments listed in the dashboard /split

### ISSUE THREAD

**Issue by phstc** [splitrb/split#603]
👋 

We've recently switched to a Redis cluster and bumped Split from 3.2.0 to 3.4.0 - is there any incompatibility with Split and a Redis cluster or in between these two versions?

After this change, we noticed that `ab_test` is switching in between control and variant and sticking to session. 

But the dashboard does not list any experiment.

![image](https://user-images.githubusercontent.com/105652/70756185-e7daaa80-1d09-11ea-8c03-08efc82f1425.png)

Testing in the Rails console 👇 

```ruby
Split.configuration.experiments # => returns the list of experiments
# {"experiment_a"=>{"alternatives"=>["control:no", "variant:yes"]}, "experiment_b"=>{"alternatives"=>["control:no", "variant:yes"
]}}
Split::ExperimentCatalog.all # => []
Split.redis.smembers(:experiments) # => []
```

Is there anything else need I can do for troubleshooting this problem?

**Comment by andrehjr** [comment]
Hi @phstc!

I'm not aware of any change between those two versions that could cause this. That's odd 🤔Because It looks like your Redis is empty. I can try using it with a Redis-cluster later.

The migration to redis-cluster happened at the same time as the split upgrade, or you only noticed that the dashboard was empty later?
Can you reproduce this issue in any other environment than your production?

**Comment by phstc** [comment]
Hi @andrehjr, thanks for the quick reply.

> you only noticed that the dashboard was empty later?

I believe I tested it before, but I can no longer tell it for sure. It is definitely switching in-between variant and control, but the dashboard `experiments` key is not getting populated.  

Is there any comment I can execute in the production console to help to debug that? 

I tried to manually execute some commands using `Split.redis` simulating the `def save` in the code, but all seems to be working. 

> Can you reproduce this issue in any other environment than your production?

Not really, but I will test it again.

**Comment by phstc** [comment]
Hi @andrehjr 

I couldn't reproduce this in development (no cluster). But today we added a new experiment to `Split.configuration.experiments` (which is loaded with a Rails initializer) and this one shows up in the dashboard, the other two are still not showing up though. 

I will do some extra debugging and let you know. But it is fine to close this for now, since there's no known issue related to that.

I will let you know if I find anything new. 

Thank you 🤜

**Comment by andrehjr** [comment]
That's odd. At least there should be `HASH` for each defined experiment. That key persisted when "ab_test" gets called.

Can you check if every experiment has the following key stored on Redis? 
`redis> HGETALL experiment_configurations/#{experiment_name}`

No worries, let me know if you find anything, I'm curious about it. 🤘

**Comment by andrehjr** [comment]
Interesting! @phstc. Does `Split.redis.exists('experiment_added_before')` returns true?

If so, let's try doing: `Split.redis.sadd('experiments', 'experiment_added_before')` and let me know if it does show up on the dashboard. Again, thanks so much for helping me debug this :)

I think we're getting close to the real issue here.

**Comment by phstc** [comment]
hm it does `exists`, and after `sadd` it showed up in the dashboard!

Thank you for helping with this issue.

**Comment by phstc** [comment]
@andrehjr something new we've just noticed. 

We added a new experiment `test_experiment_remove_me` for testing the issue above.

![image](https://user-images.githubusercontent.com/105652/71268314-a86c1980-231a-11ea-81ee-877221302404.png)

Then we removed it, it's no longer listed in `Split.configuration.experiments` but it is still showing up on the dashboard.  Even if I delete it in the dashboard it eventually shows up again.

**Comment by chendo** [comment]
I've run into the same problem here, on Split 4.0.1. However, we did not upgrade, but suspect some data loss either through evictions in Redis.

As OP reported, `Split.configuration.experiments` shows all experiments, but `Split::ExperimentCatalog.all` only shows the one we added recently.

Doing `Split.redis.sadd(:experiments, "<missing experiment>")` causes the missing experiment to show up again, and all the data is there!
